odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -234,6 +234,28 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 dest_record_id=dst_record.id,
             ))
 
+            # company dependent fields referring the merged records might not be set on the table,
+            # we need to update the default values
+            self.env.cr.execute(SQL(
+                """
+                UPDATE ir_default
+                SET json_value =
+                    CASE
+                        WHEN json_value::int IN %(src_record_ids)s
+                        THEN %(dest_record_id)s
+                        ELSE json_value::int
+                    END
+                WHERE
+                    field_id IN (
+                        SELECT id FROM ir_model_fields WHERE name = %(field)s
+                    )
+                AND json_value ~ '^[0-9]$';
```

Comment:
not sure if we can trust the ir.default data
if the fallback in the ir_default is not `json_value ~ '^[0-9]$'` when read the value it will still fail since it cannot be parsed to integer when read

Reply:
> not sure if we can trust the ir.default data
WDYM?
> if the fallback in the ir_default is not json_value ~ '^[0-9]$' when read the value it will still fail since it cannot be parsed to integer when read
With this condition we only update value that can be parsed into integer, we won't have error as non digit values will be skipped. I don't see a case when we would like to update non digit values when merging records, since we're trying to set the `dest_record_id` like in the above request

Reply:
I see, it is mainly for `'false'` and `'null'` for the `json_value`. so you can keep it